### PR TITLE
fix(path): stop at / in stem()

### LINKED ISSUE #607: fix(internal/path): `ext` and `stem` behavior in edge cases.
related discussion: https://github.com/moonbitlang/maria/pull/606#discussion_r2548666886

- [x] ext function is fixed in https://github.com/moonbitlang/maria/pull/608
- [ ] stem function currently work in https://github.com/moonbitlang/maria/pull/615

### PR DESCRIPTION
**Pull request by tonyfettes**
Closes #607

### CODE REVIEW

File: internal/path/path.mbt [hunk 1]
```diff
@@ -65,13 +65,26 @@ pub fn ext(path : StringView) -> StringView {
 
 ///|
 /// Extract stem part exclude `.`
-/// 
-/// property: `stem(path) + "." + ext(path)`
+///
+/// property: `stem(path) + ext(path) == path`
```
Comment (tonyfettes) [inline internal/path/path.mbt]: /cc @illusory0x0 updated the property holds between the `stem` and `ext` function

File: internal/path/path_test.mbt [hunk 2]
```diff
@@ -137,5 +149,5 @@ test "choose second if absolute" {
 test "ext/stem:edge cases" {
   let path = "proj.tar/file"
   inspect(@path.ext(path), content="")
-  inspect(@path.stem(path), content="proj")
+  inspect(@path.stem(path), content="proj.tar/file")
```
[the remark this answers is not in the document]
Comment (tonyfettes) [reply internal/path/path_test.mbt]: Intended behavior, as some callsites depend on the behavior of `stem` that it does not strip the directory components in the path.

**Comment by illusory0x0** [comment]
In order to fix stem function incorrect behavior, we use following steps.

1. mark some call sites rely on incorrect behavior: https://github.com/moonbitlang/maria/pull/625
2. add replace_ext: https://github.com/moonbitlang/maria/issues/622
3. refactor above call sites via replace_ext
4. fix stem function make it consistent with Python `pathlib` name convention.  `internal/moon` depend on this behavior.
